src/fs/driver/fat/fat_common.c

```diff
@@ -940,10 +956,10 @@ int fat_root_dir_record(void *bdev) {
 		return -1;
 	}
 
-	cluster = fat_get_free_fat_(&fsi, fat_sector_buff);
+	cluster = 0; //fat_get_free_fat_(&fsi, fat_sector_buff);
```

Comment:
Probably, that's related to fact, that for FAT12 size of cluster entry (12 bits) is not multiple of byte size (8 bits), so three bytes store info about 2 clusters (12 bit for each cluster, 8 bit per byte), and align is required.